src/cpl/share_esmf/cropcalStreamMod.F90

```diff
@@ -0,0 +1,431 @@
+         stream_fldFileName_sdate,     &
+         stream_fldFileName_cultivar_gdds, &
+         stream_meshfile_cropcal
+
+    ! Default values for namelist
+    stream_year_first_cropcal  = 1      ! first year in stream to use
+    stream_year_last_cropcal   = 1      ! last  year in stream to use
+    model_year_align_cropcal   = 1      ! align stream_year_first_cropcal with this model year
+    stream_meshfile_cropcal    = ''
+    stream_fldFileName_sdate = ''
+    stream_fldFileName_cultivar_gdds = ''
+    ! Will need modification to work with mxsowings > 1
+    ncft = mxpft - npcropmin + 1 ! Ignores generic crops
+    allocate(stream_varnames_sdate(ncft))
+    allocate(stream_varnames_cultivar_gdds(ncft))
+    do n = 1,ncft
+       ivt = npcropmin + n - 1
+       write(stream_varnames_sdate(n),'(a,i0)') "sdate1_",ivt
+       write(stream_varnames_cultivar_gdds(n),'(a,i0)') "gdd1_",ivt
+    end do
+
+    ! Read cropcal_streams namelist
+    if (masterproc) then
+       open( newunit=nu_nml, file=trim(NLFilename), status='old', iostat=nml_error )
+       call find_nlgroup_name(nu_nml, 'cropcal_streams', status=nml_error)
+       if (nml_error == 0) then
+          read(nu_nml, nml=cropcal_streams,iostat=nml_error)
+          if (nml_error /= 0) then
+             call endrun(subname // ':: ERROR reading cropcal_streams namelist')
+          end if
+       else
+          call endrun(subname // ':: ERROR finding cropcal_streams namelist')
+       end if
+       close(nu_nml)
+    endif
+    call shr_mpi_bcast(stream_year_first_cropcal  , mpicom)
+    call shr_mpi_bcast(stream_year_last_cropcal   , mpicom)
+    call shr_mpi_bcast(model_year_align_cropcal   , mpicom)
+    call shr_mpi_bcast(stream_fldFileName_sdate   , mpicom)
+    call shr_mpi_bcast(stream_fldFileName_cultivar_gdds, mpicom)
+    call shr_mpi_bcast(stream_meshfile_cropcal    , mpicom)
+
+    if (masterproc) then
+       write(iulog,*)
+       write(iulog,*) 'cropcal_stream settings:'
+       write(iulog,'(a,i8)') '  stream_year_first_cropcal  = ',stream_year_first_cropcal
+       write(iulog,'(a,i8)') '  stream_year_last_cropcal   = ',stream_year_last_cropcal
+       write(iulog,'(a,i8)') '  model_year_align_cropcal   = ',model_year_align_cropcal
+       write(iulog,'(a,a)' ) '  stream_fldFileName_sdate   = ',trim(stream_fldFileName_sdate)
+       write(iulog,'(a,a)' ) '  stream_fldFileName_cultivar_gdds   = ',trim(stream_fldFileName_cultivar_gdds)
+       write(iulog,'(a,a)' ) '  stream_meshfile_cropcal    = ',trim(stream_meshfile_cropcal)
+       do n = 1,ncft
+          write(iulog,'(a,a)' ) '  stream_varnames_sdate  = ',trim(stream_varnames_sdate(n))
+          write(iulog,'(a,a)' ) '  stream_varnames_cultivar_gdds  = ',trim(stream_varnames_cultivar_gdds(n))
+       end do
+       write(iulog,*)
+    endif
+
+    use_cropcal_rx_sdates = stream_fldFileName_sdate /= ''
+    use_cropcal_rx_cultivar_gdds = stream_fldFileName_cultivar_gdds /= ''
```

Comment:
Thinking about it some more ([previously](https://github.com/ESCOMP/CTSM/pull/1863#discussion_r1007248347)), I do think I'm going to remove the default files, thus requiring users to provide their own. If it's ever decided for this to be scientifically supported or whatever, then I can do the work to provide defaults at half-degree resolution (the native resolution of the GGCMI crop calendars). I'd also then make the changes you suggest here.

I'll let you mark this as resolved if you agree with that plan; otherwise let me know.

Reply:
Okay, I'm fine with your plan to make the default be empty and require the user to provide their own files in order to enable this behavior. So I'll check this off.